src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.StaticPipe;
+using Microsoft.ML.StaticPipe.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// This is a helper class that is required to use the <see cref="DnnImageFeaturizerEstimator"/>.
+    /// Note that by default, it is not usable as it does not have any valid methods that return an <see cref="EstimatorChain{TLastTransformer}"/>
+    /// that is used by the DnnImageFeaturizeEstimator.
+    /// In order to use this, at least one model project with the corresponding extension methods must by included.
+    /// See Microsoft.ML.DNNImageFeaturizer.ResNet18 for an example.
+    /// </summary>
+    public sealed class DnnImageModelSelector
```

Comment:
Shouldn't this be named "Catalog" and hang off the MLContext, like the rest of the objects we want to have extension methods on?

/cc @Zruty0 @TomFinley

Reply:
See for example [https://github.com/dotnet/machinelearning/blob/d68388a1c9994a5b429b194b64b2b0782834cb78/src/Microsoft.ML.Data/Transforms/TransformsCatalog.cs#L11](https://github.com/dotnet/machinelearning/blob/d68388a1c9994a5b429b194b64b2b0782834cb78/src/Microsoft.ML.Data/Transforms/TransformsCatalog.cs#L11)

---
In reply to: [232778442](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232778442) [](ancestors = 232778442)